src/ai/backend/manager/models/image.py

```diff
@@ -1011,31 +1020,42 @@ def _is_image_accessible_for_user(
         if image.registry not in allowed_registries:
             return False
         labels = cast(dict[str, str], image.labels)
-        if labels.get("ai.backend.customized-image.owner") != f"user:{user_id}":
+        owner_label = labels.get("ai.backend.customized-image.owner")
+        if owner_label is not None and owner_label != f"user:{user_id}":
             return False
         return True
 
     async def _in_user_scope(
         self,
         ctx: ClientContext,
         scope: UserScope,
+        has_admin_privilege: bool = False,
     ) -> ImagePermissionContext:
         allowed_registries = await self._get_allowed_registries_for_user(ctx, scope.user_id)
 
         permissions = await self.calculate_permission(ctx, scope)
         image_id_permission_map: dict[UUID, frozenset[ImagePermission]] = {}
 
-        _img_query_stmt = (
+        img_query_stmt = (
             sa.select(ImageRow)
             .join(ImageRow.registry_row)
             .options(load_only(ImageRow.id, ImageRow.labels, ImageRow.registry))
             .where(ContainerRegistryRow.is_global == true())
         )
 
-        for row in await self.db_session.scalars(_img_query_stmt):
+        for row in await self.db_session.scalars(img_query_stmt):
             image_row = cast(ImageRow, row)
-            if not self._is_image_accessible_for_user(image_row, allowed_registries, scope.user_id):
-                continue
+            if has_admin_privilege:
+                if image_row.registry not in allowed_registries:
+                    continue
+            else:
+                if not image_row.is_customized_by(
+                    scope.user_id
+                ) or not self._is_image_accessible_for_user(
+                    image_row, allowed_registries, scope.user_id
+                ):
+                    continue
```

Comment:
The logic is verbose, complex, and contains many redundant parts, so it seems to need refactoring.
Rather than reorganizing the if conditions, it would be better to rewrite the function.